core/src/components/checkbox/checkbox.tsx

```diff
@@ -181,6 +181,15 @@ export class Checkbox implements ComponentInterface {
     this.ionBlur.emit();
   };
 
+  private onKeyDown = (ev: KeyboardEvent) => {
+    if (ev.key === ' ' || ev.key === 'Enter') {
```

Comment:
Based on [W3C](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/examples/checkbox/#kbd_label), only the `SPACE` key should toggle. I also verified with Chakra UI, a library known for it's accessibility.

```suggestion
    if (ev.key === ' ') {
```